docs: remove extraneous parts of DHIS2 Docker tutorial

radnov opened the pull request:
This PR is a proposal to simplify the DHIS2 Docker tutorial and remove all extraneous parts like updating your system, installing Node, Docker, Nginx, etc. All of the removed steps seem too opinionated and don't apply to all users, so we should probably just provide links to external documentation instead.

As far as running DHIS2 in Docker goes, we should probably also mention running it via `docker compose` (like described [here](https://github.com/dhis2/dhis2-core/#run-dhis2-in-docker)) and not only the `d2 cluster` tool, as it might be easier for some users, depending on their use case.

teleivo commented:
> This PR is a proposal to simplify the DHIS2 Docker tutorial and remove all extraneous parts like updating your system, installing Node, Docker, Nginx, etc. All of the removed steps seem too opinionated and don't apply to all users, so we should probably just provide links to external documentation instead.

Is a good idea as Docker will likely keep its docs on how to install it on any imaginable system up to date 😋

teleivo commented on docs/tutorials/dhis2-docker.md:
```diff
@@ -3,312 +3,12 @@ id: dhis2-docker
-every month or so.
 
 ## Install D2 CLI
 
```
I think we should always just refer to the "most authoritative" source in order for us to have docs that are up-to-date.

I suggest we add a "Prerequisites" section. Maybe

```markdown
## Prerequisites

Make sure you have the following tools installed

* [Docker](https://www.docker.com/)
* [Docker Compose](https://docs.docker.com/compose/)
* [DHIS2 d2 CLI](https://cli.dhis2.nu/#/getting-started) - only needed if you want to use `d2` for setting up DHIS2 in Docker.
```

We can also move the link to `d2` at the top of a `d2` section. In case we want to split the docs into `Docker Compose` and `d2`.

radnov commented:
@teleivo @Topener I simplified the dhis2-docker tutorial quite a bit by linking external documentation instead of repeating it and updated the reference in the quickstart.

Need help with the following:
1. Should we update the contents of the [DHIS2 and Docker blog post](https://developers.dhis2.org/blog/2019/10/dhis2-and-docker) or maybe just add a link back to the dhis2-docker tutorial and say that the blog is outdated?
2. Should this part specific to the d2 CLI be moved to the d2 CLI docs? Not sure why it's not already mentioned there.

Let me know what you think. 👀

teleivo commented on docs/quickstart.md:
```diff
@@ -64,12 +46,12 @@ d2 cluster up 2.39.0.1 --db-version 2.39 --seed
 
 ### Next steps
 
 Now that you havea DHIS2 instance you can put it to the test by developing DHIS2 apps.
```
```suggestion
Now that you have a DHIS2 instance you can put it to the test by developing DHIS2 apps.
```

teleivo commented on docs/tutorials/dhis2-docker.md:
````diff
@@ -1,315 +1,43 @@
 ```
 
-### Control the Docker service
+This is the preferred way if you already have the `dhis2-core` repository cloned and you want a simple setup that consist of DHIS2 and a database.
````
```suggestion
This is the preferred way if you want a simple setup that consist of DHIS2 and a database. You can either clone the `dhis2-core` repository or download only the `[docker-compose](https://github.com/dhis2/dhis2-core/blob/master/docker-compose.yml)` file.
```

teleivo commented on docs/tutorials/dhis2-docker.md:
```diff
@@ -1,315 +1,43 @@
-every month or so.
+---
 
+[//]: # (TODO Maybe the following should be moved to the D2 CLI docs)
```
I think it should be removed as its in https://cli.dhis2.nu/#/getting-started
and in https://developers.dhis2.org/docs/tutorials/setup-env 😅https://cli.dhis2.nu/#/getting-started

teleivo commented on blog/2019-10-29-dhis2-and-docker.md:
```diff
@@ -7,6 +7,9 @@ author_image_url: https://avatars.githubusercontent.com/u/12664326?s=400&v=4
 tags: [docker]
 ---
 
+[//]: # (TODO Should we mention an update date?)
```
The blog post is from `October 29, 2019` which does not seem recent to me. Maybe it should not show up in https://developers.dhis2.org/blog recent posts.

Not sure if I would update a blog post. To me that is a one time "article". If there is information that we discover isn't true I would update it. But I would not keep a blog post up to date for 5 years or more 😅 

We should extract information that is useful and still applies to the Docker section in https://github.com/dhis2/dhis2-core.

radnov replied:
@teleivo I added an update note to the top of the blog post - https://github.com/dhis2/developer-portal/pull/105/commits/f5933569dd31327795fe8d8459478e7acfb7a291

And extracted some of the info to the dhis2-core readme - https://github.com/dhis2/dhis2-core/pull/12831

Do you think we should extract something more?

teleivo replied:
Adding that comment to the blog post makes sense. I would also remove the blog post from recent posts 😄  Its not recent and outdated. @Topener whats your opinion on the above?

Topener replied:
The recent blogposts are indeed not recent, but it's the last 15 (or so) articles published there. One way to get rid of it, is writing more (I'm working on it).

I wouldn't update the old blogpost with new content, but a warning it's outdated is for sure a good one! Preferably with a link to new content in that case, either a guide or a new blogpost

teleivo replied:
@Topener has done what you suggested. Is this change ready to be merged from your point of view?

tonsV2 reviewed (approved):
I don't anything to add, I really like that we're keeping short and concise

teleivo commented on docs/tutorials/dhis2-docker.md:
````diff
@@ -1,384 +1,41 @@
-```
-sudo apt-get autoremove
-```
+## DHIS2 in Docker in production
````
nit: newline after titles

teleivo commented on docs/tutorials/dhis2-docker.md:
````diff
@@ -1,384 +1,41 @@
-sudo apt-get autoremove
-```
+## DHIS2 in Docker in production
+We can't recommend using [the DHIS2 Docker images](https://github.com/dhis2/dhis2-core#pre-built-images) in production yet. At this point we don’t have enough experience and therefore can’t vouch for their stability for “mission critical” production use. We are not saying that someone experienced with running Docker in production shouldn’t use them.  
````
This looks like the "old" sentence. Should we use the updated one that we iterated on in slack the other day?

Topener reviewed (changes requested):
Only one small suggestion, other than that it looks good

Topener reviewed (approved):
LGTM
